### DISCUSSION TITLE
Bug while using .env.development for providing accessToken

### DISCUSSION THREAD

**Discussion by Officer-R** [gatsbyjs/gatsby#29203] [Help]
@hashimwarren I am struggling with a bug while using contentful access token via dotenv file.

I am attaching the screenshots below can you or anyone in the community point me out what's wrong ?

![Screenshot 2021-01-26 at 6 42 02 PM](https://user-images.githubusercontent.com/37110560/105849484-768db200-6006-11eb-9f6a-37febbbd35bc.jpg)
![Screenshot 2021-01-26 at 6 41 42 PM](https://user-images.githubusercontent.com/37110560/105849500-7d1c2980-6006-11eb-9dea-bb18cabcd707.jpg)

**Comment by suyundukov** [comment]
Hey, @rohit1101. Have you loaded your environment variables from `.env.devlopment` inside `gatsby.config.js`? Project environment variables that you defined in the `.env.*` files will not be available in `gatsby-config.js`, unless you loaded them manually.

**Reply by Officer-R** [reply]
No how do I load env variables from `.env.development` in `gatsby.config.js` ?

**Reply by suyundukov** [reply]
Add the following at the beginning of `gatsby-config.js`:

```js
require("dotenv").config({
  path: `.env.${process.env.NODE_ENV}`,
})
```

For more info: [https://gatsby.dev/env-vars](https://www.gatsbyjs.com/docs/how-to/local-development/environment-variables/#server-side-nodejs)

**Reply by Officer-R** [reply]
Ya googled it a bit and figured it out bro thanks for the help anyways..

How's your progress with Gatsby till now ?

**Reply by hkalic** [reply]
> Ya googled it a bit and figured it out bro thanks for the help anyways..
> 
> How's your progress with Gatsby till now ?

Can you share how did you resolved it?

I can't access `process.env.GATSBY_CONTENTFUL_ACCESS_TOKEN` inside `gatsby-config.js`. Error is `Invalid plugin options for "gatsby-source-contentful"`.
If I enter it manually (without .env.development) it works, of course. Later on a random page, I use `{${process.env.GATSBY_CONTENTFUL_ACCESS_TOKEN}}` which is shown correctly.

I read [https://gatsby.dev/env-vars](url) like 10 times and can't figure it out.

**Reply by Officer-R** [reply]
@hkalic Ya it's a simple stuff.

![Screenshot 2021-03-28 at 10 42 43 PM](https://user-images.githubusercontent.com/37110560/112761091-1374c780-9017-11eb-913a-f2821e215c88.jpg)
This is how your gatsby.config.js should start.

Then make sure the `.env.development` looks like this:

`GATSBY_CONTENTFUL_ACCESS_TOKEN=Your_Token` 

Hope this helps you.If you are still stuck reply here.

**Reply by hkalic** [reply]
> @hkalic Ya it's a simple stuff.
> 
> ![Screenshot 2021-03-28 at 10 42 43 PM](https://user-images.githubusercontent.com/37110560/112761091-1374c780-9017-11eb-913a-f2821e215c88.jpg)
> This is how your gatsby.config.js should start.
> 
> Then make sure the `.env.development` looks like this:
> 
> `GATSBY_CONTENTFUL_ACCESS_TOKEN=Your_Token`
> 
> Hope this helps you.If you are still stuck reply here.

**Thanks for the help**.
I was trying to do it ES6 way with (in the `gatsby-config.js`)

`import dotenv from "dotenv"`
`dotenv.config( {path: ".env.${process.env.NODE_ENV}"} )`

... so, I get `SyntaxError: Cannot use import statement outside a module`
I know it's a noob mistake but can't help myself.

Finally, I did exactly what you said and it works. As I see Gatsby uses ES modules but `gatsby-config.js` is a common JS module.
